src/Logging/Logging.Console/src/AnsiLogConsole.cs

```diff
@@ -23,26 +26,39 @@ public AnsiLogConsole(IAnsiSystemConsole systemConsole)
         public void Write(string message, ConsoleColor? background, ConsoleColor? foreground)
         {
             // Order: backgroundcolor, foregroundcolor, Message, reset foregroundcolor, reset backgroundcolor
+            bool backgroundColorChanged = false;
+            bool foregroundColorChanged = false;
+
             if (background.HasValue)
             {
-                _outputBuilder.Append(GetBackgroundColorEscapeCode(background.Value));
+                string backgroundColor = GetBackgroundColorEscapeCode(background.Value);
+                if (_defaultBackgroundColor != backgroundColor)
+                {
+                    backgroundColorChanged = true;
+                    _outputBuilder.Append(backgroundColor);
+                }
             }
 
             if (foreground.HasValue)
             {
-                _outputBuilder.Append(GetForegroundColorEscapeCode(foreground.Value));
+                string foregroundColor = GetForegroundColorEscapeCode(foreground.Value);
+                if (_defaultForegroundColor != foregroundColor)
+                {
+                    foregroundColorChanged = true;
+                    _outputBuilder.Append(foregroundColor);
+                }
             }
 
             _outputBuilder.Append(message);
 
-            if (foreground.HasValue)
+            if (foregroundColorChanged)
             {
-                _outputBuilder.Append("\x1B[39m\x1B[22m"); // reset to default foreground color
+                _outputBuilder.Append(_defaultForegroundColor); // reset to default foreground color
             }
 
-            if (background.HasValue)
+            if (backgroundColorChanged)
             {
-                _outputBuilder.Append("\x1B[49m"); // reset to the background color
+                _outputBuilder.Append(_defaultBackgroundColor); // reset to the background color
```

Comment:
```suggestion
                _outputBuilder.Append(_defaultBackgroundColor); // reset to default background color
```

I see you've taken the comment from the code before, but it should be aligned to the foreground-case.